From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dreeam <61569423+Dreeam-qwq@users.noreply.github.com>
Date: Mon, 15 Jan 2024 10:53:10 -0500
Subject: [PATCH] Reduce active items finding hopper nearby check

This patch add a toggle for active items checking MinecraftHopper nearby,

But still recommend to turn-off `checkForMinecartNearItemWhileActive`
Since `Reduce-hopper-item-checks.patch` will cause lag under massive dropped items

diff --git a/net/minecraft/world/entity/item/ItemEntity.java b/net/minecraft/world/entity/item/ItemEntity.java
index 2b6982d3adeed66cba80f089ba015db25898fcf7..c1e7c04b048099dab7bb4cabdcb53add2201247f 100644
--- a/net/minecraft/world/entity/item/ItemEntity.java
+++ b/net/minecraft/world/entity/item/ItemEntity.java
@@ -234,7 +234,11 @@ public class ItemEntity extends Entity implements TraceableEntity {
                 this.discard(org.bukkit.event.entity.EntityRemoveEvent.Cause.DESPAWN); // CraftBukkit - add Bukkit remove cause
                 return; // Gale - EMC - reduce hopper item checks
             }
-            this.markNearbyHopperCartsAsImmune(); // Gale - EMC - reduce hopper item checks
+            // Leaf start - Reduce active items finding hopper nearby check
+            if (level().galeConfig().smallOptimizations.reducedIntervals.checkNearbyItem.hopper.minecart.temporaryImmunity.checkForMinecartNearItemWhileActive) {
+                this.markNearbyHopperCartsAsImmune(); // Gale - EMC - reduce hopper item checks
+            }
+            // Leaf end - Reduce active items finding hopper nearby check
         }
     }
 
